src/api/app/helpers/webui/staging/project_helper.rb

```diff
@@ -0,0 +1,9 @@
+module Webui::Staging::ProjectHelper
+  def icon_for_checks(checks, missing_checks)
+    return 'fa-eye text-info' if missing_checks.present?
+    return 'fa-check-circle text-primary' if checks.blank?
+    return 'fa-eye text-info' if checks.any? { |check| check.state == 'pending' }
+    return 'fa-check-circle text-primary' if checks.all? { |check| check.state == 'success' }
```

Comment:
so far we checked for failed? first, but your version would work as well - but please add a success? function to check and use it here